rem/DB.swift

```diff
@@ -215,76 +225,98 @@ class DatabaseManager {
 
-        var results: [(Int64, String, String?, Date)] = []
+        var results: [(Int64, String, String?, Date, String, Int64)] = []
         do {
             for row in try db.prepare(query) {
                 let frameId = row[allText[frameId]]
                 let matchedText = row[self.text]
                 let applicationName = row[frames[activeApplicationName]]
                 let timestamp = row[frames[timestamp]]
-                results.append((frameId, matchedText, applicationName, timestamp))
+                let filePath = row[videoChunks[filePath]]
+                let offsetIndex = row[frames[offsetIndex]]
+                results.append((frameId, matchedText, applicationName, timestamp, filePath, offsetIndex))
             }
         } catch {
             print("Search error: \(error)")
         }
         return results
     }
     
-    func getRecentResults(limit: Int = 9) -> [(frameId: Int64, fullText: String?, applicationName: String?, timestamp: Date)] {
-        let query = allText
-            .join(frames, on: frames[id] == allText[frameId])
+    func getRecentResults(limit: Int = 9, offset: Int = 0) -> [(frameId: Int64, fullText: String?, applicationName: String?, timestamp: Date, filePath: String, offsetIndex: Int64)] {
+        let query = frames
+            .join(videoChunks, on: frames[chunkId] == videoChunks[id])
+            .select(frames[id], frames[activeApplicationName], frames[timestamp], videoChunks[filePath], frames[offsetIndex])
             .order(frames[timestamp].desc)
-            .limit(limit)
-            .select(allText[frameId], frames[activeApplicationName], frames[timestamp])
+            .limit(limit, offset: offset)
 
-        var results: [(Int64, String?, String?, Date)] = []
+        var results: [(Int64, String?, String?, Date, String, Int64)] = []
         do {
             for row in try db.prepare(query) {
-                let frameId = row[allText[frameId]]
+                let frameId = row[frames[id]]
                 let applicationName = row[frames[activeApplicationName]]
                 let timestamp = row[frames[timestamp]]
-                results.append((frameId, nil, applicationName, timestamp))
+                let filePath = row[videoChunks[filePath]]
+                let offsetIndex = row[frames[offsetIndex]]
+                results.append((frameId, nil, applicationName, timestamp, filePath, offsetIndex))
             }
         } catch {
             print("Error fetching recent results: \(error)")
         }
         return results
     }
     
-    func getImage(index: Int64) -> CGImage? {
+    func getImage(index: Int64, maxSize: CGSize? = nil) -> CGImage? {
         guard let frameData = DatabaseManager.shared.getFrame(forIndex: index) else { return nil }
         
         let videoURL = URL(fileURLWithPath: frameData.filePath)
-        if !FileManager.default.fileExists(atPath: videoURL.path) {
-            return nil
-        }
```

Comment:
omg file io ON EVERY SINGLE SEARCH RESULT

someone else take over this project